src/website/utils/url_utils.ts

```diff
@@ -0,0 +1,186 @@
+
+function urlSearchParamsToParams(searchParams: URLSearchParams): Params {
+  const result: Params = {};
+  for (const [key, value] of searchParams.entries()) {
+    if (paramKeys.includes(key)) {
+      result[key] = value;
+    }
+  }
+  return result;
+}
+
+const decodeURL = (url: string): string => {
+  const decodedUrl = decodeURIComponent(url);
+  return decodedUrl.endsWith("/") ? decodedUrl.slice(0, -1) : decodedUrl;
+};
+
+/** Try to parse a `string` as a list of 2 or more URLs. Returns `undefined` if the string is not a valid URL list. */
+const tryDecodeURLList = (url: string, delim = ","): string[] | undefined => {
+  if (!url.includes(delim)) {
+    return undefined;
+  }
+
+  const urls = url.split(delim).map((u) => decodeURL(u));
+
+  // Verify that all urls are valid
+  for (const u of urls) {
+    try {
+      new URL(u);
+    } catch (_e) {
+      return undefined;
+    }
+  }
+
+  return urls;
+};
+
+async function loadDataset(dataset: string, id: string): Promise<Partial<AppProps>> {
+  const db = new FirebaseRequest();
+  const args: Partial<AppProps> = {};
+
+  const datasets = await db.getAvailableDatasets();
+
+  let datasetMeta: DatasetMetaData | undefined = undefined;
+  for (const d of datasets) {
+    const innerDatasets = d.datasets!;
+    const names = Object.keys(innerDatasets);
+    const matchingName = names.find((name) => name === dataset);
+    if (matchingName) {
+      datasetMeta = innerDatasets[matchingName];
+      break;
+    }
+  }
+  if (datasetMeta === undefined) {
+    console.error(`No matching dataset: ${dataset}`);
+    return {};
+  }
+
+  const datasetData = await db.selectDataset(datasetMeta.manifest!);
+  const baseUrl = datasetData.volumeViewerDataRoot + "/";
+  args.imageDownloadHref = datasetData.downloadRoot + "/" + id;
+  // args.fovDownloadHref = datasetData.downloadRoot + "/" + id;
+
+  const fileInfo = await db.getFileInfoByCellId(id);
+  args.imageUrl = baseUrl + fileInfo!.volumeviewerPath;
+  args.parentImageUrl = baseUrl + fileInfo!.fovVolumeviewerPath;
+
+  return args;
+}
+
+export async function getArgsFromParams(urlSearchParams: URLSearchParams): Promise<{
+  args: Partial<AppProps>;
+  viewerSettings: Partial<GlobalViewerSettings>;
+}> {
+  const params = urlSearchParamsToParams(urlSearchParams);
+  let args: Partial<AppProps> = {};
+  const viewerSettings: Partial<GlobalViewerSettings> = {};
+
+  if (params) {
+    if (params.mask) {
+      viewerSettings.maskAlpha = parseInt(params.mask, 10);
+    }
+    if (params.view) {
+      const mapping = {
+        "3D": ViewMode.threeD,
+        Z: ViewMode.xy,
+        Y: ViewMode.xz,
+        X: ViewMode.yz,
+      };
+      const allowedViews = Object.keys(mapping);
+      let view: "3D" | "X" | "Y" | "Z";
+      if (allowedViews.includes(params.view)) {
+        view = params.view as "3D" | "X" | "Y" | "Z";
+      } else {
+        view = "3D";
+      }
+      viewerSettings.viewMode = mapping[view];
```

Comment:
I did make a minor change here; @frasercl I imagine there's cleaner way to do this